---
title: Biogeography and Environmental Conditions Shape Phage and Bacteria Interaction Networks Across the Healthy Human Microbiome
author: 
- name: Geoffrey D Hannigan
  affiliation: 1
- name: Melissa B Duhaime
  affiliation: 2
- name: Danai Koutra
  affiliation: 3
- name: Patrick D Schloss
  affiliation: 1,\*
address:
  - code: 1
    address: Department of Microbiology & Immunology, University of Michigan, Ann Arbor, Michigan, 48109
  - code: 2
    address: Department of Ecology and Evolutionary Biology, University of Michigan, Ann Arbor, Michigan, 48109
  - code: 3
    address: Department of Computer Science, University of Michigan, Ann Arbor, Michigan, 48109
  - code: \*
    address: To whom correspondence may be addressed.
output:
  md_document:
    variant: markdown
fontsize: 11pt
mainfont: 'Helvetica'
---

&nbsp;

&nbsp;

&nbsp;

| ***Corresponding Author Information***
| Patrick D Schloss, PhD
| 1150 W Medical Center Dr. 1526 MSRB I
| Ann Arbor, Michigan 48109
| Phone: (734) 647-5801
| Email: pschloss@umich.edu

***Running Title***: Network Diversity of the Healthy Human Microbiome

***Journal***: Genome Research ([*Preparation Details*](http://genome.cshlp.org/site/misc/ifora_mspreparation.xhtml))

***Keywords***: Virome, Microbiome, Graph Theory, Machine Learning

***Text Length***: 35,640 / 50,000 Characters

\* *Figures at the end of the document for internal review only.*

\newpage

```{r loadfiles, echo=FALSE}
sigfig = 3
genfigurestats <- read.delim(file = "../rtables/genfigurestats.tsv", header = TRUE, sep = "\t")
nestats <- read.delim(file = "../rtables/nestats.tsv", header = TRUE, sep = "\t")
genmodelper <- read.delim(file = "../rtables/genmodelper.tsv", header = TRUE, sep = "\t")
interstats <- read.delim(file = "../rtables/interstats.tsv", header = TRUE, sep = "\t")
skinboxsig <- read.delim(file = "../rtables/skinboxsig.tsv", header = TRUE, sep = "\t")
```

# Abstract
Viruses and bacteria are critical components of the human microbiome and play important roles in health and disease. Previous work has relied on studying bacteria and phages in isolation, reducing them to two separate communities. These approaches have not captured how these communities interact to share information (e.g. horizontal gene transfer) and maintain stability. We developed and implemented a network-based analytical approach to provide an initial understanding of phage-bacteria network diversity throughout the human body. We accomplished this by building a machine learning algorithm to predict which phages infected which bacteria in a given microbiome. This algorithm was applied to paired viral and bacterial metagenomic sequence sets from three previously published human cohorts. We organized the predicted interactions into networks that allowed us to evaluate the diversity of phage-bacteria connectedness across the human body. We focused on establishing a foundational understanding of microbiome network structures across the human body. We found that gut and skin network structure was person-specific, and was not conserved among cohabitating family members. High fat diets and obesity were associated with less connected networks. There were significant differences in network structure between skin sites, with those exposed to the external environment being less connected and more prone to instability. Altogether this study characterizes the diversity of microbiome networks across the human body, and provides a baseline for future studies to investigate the role of ecological networks in disease states.

**Word Count**: 236 / 250

\newpage

# Introduction

Viruses and bacteria are critical components to the human microbiome and play important roles in health and disease. Bacterial communities have been associated with diseases including a wide range of skin conditions [@Hannigan:2013im], acute and chronic wound healing conditions [@Hannigan:2014be; @Loesche:2016ev], and gastrointestinal diseases including inflammatory bowel disease [@He:2016ch; @Norman:2015kb], *Clostridium difficile* infections [@Seekatz:2016fz], and colorectal cancer [@Zackular:2014fba; @Baxter:2014hb]. Altered viromes (virus communities consisting primarily of bacteriophages in humans) have also been associated with various diseases and environmental perturbations including inflammatory bowel disease [@Norman:2015kb; @Manrique:2016dx], periodontal disease [@Ly:2014ew], spread of antibiotic resistance [@Modi:2013fia; @Hannigan:2015fz], and others [@Monaco:2016ita; @Hannigan:2015fz; @Minot:2011ez; @SantiagoRodriguez:2015gd; @Abeles:2015dy; @Abeles:2014kj]. These shifts in virus community composition are not reflections of the bacterial communities, but rather act in concert with bacteria as one single overall community [@Norman:2015kb; @Haerter:2014ii]. Individual phages within these communities are capable of lysing their bacterial hosts and modulating their functionality through horizontal gene transfer (i.e. transduction) and altering bacterial host gene expression.

Previous work has relied on studying bacterial and phage communities in isolation, reducing them to two separate communities. In reality, bacteria and phage communities are dynamic and complex, acting in concert to share genetic information and to maintain stable ecosystems, and removal of members can disrupt or even collapse those ecosystems [@Haerter:2014ii; @Harcombe:2005fd; @Middelboe:2001fl; @Poisot:2011jc; @Thompson:2012ki; @Moebus:1981kp; @Flores:2013hc; @Poisot:2012fh; @Poisot071993; @Flores:2011bh; @Jover:2015ev]. The conventional approach to understanding the human microbiome is unable to capture how these communities interact and influence each other, but relationship-based network approaches allow us to capture this information. We aimed to bridge this knowledge gap by characterizing human bacterial and phage communities by their *relationships*, leveraging machine learning and graph theory techniques. We focused on characterizing the network diversity of the healthy human microbiome (including the skin and gut), so as to provide a foundation on which we and others can build further studies into the network dynamics associated with disease states.

We began characterizing human microbiome phage and bacteria networks by leveraged three published microbiome datasets with paired virus and bacterial metagenomic sequence sets [@Hannigan:2015fz; @Minot:2011ez; @Reyes:2010cwa; @Turnbaugh:2009ei]. These datasets included gut and skin samples, allowing us to gain broader insights into human microbiome network diversity across different body sites. Our approach built off of previous large-scale phage-bacteria microbiome network analyses by inferring interactions using metagenomic datasets, instead of using culture-based techniques [@Moebus:1981kp; @Flores:2013hc]. Our metagenomic interaction inference model was powered beyond previous models by its inclusion of protein interaction data, inclusion of negative interactions as well as positive, and the use of a more sophisticated machine learning algorithm instead of linear models [@Edwards:2015iz].

Using this approach, we were able to go beyond contemporary isolated methods to provide a basic understanding of the community network dynamics associated with healthy human phages and bacteria. By building and utilizing a microbiome network, we found that different people, body sites, and anatomical locations not only support distinct microbiome membership and diversity [@Hannigan:2015fz; @Minot:2011ez; @Reyes:2010cwa; @Turnbaugh:2009ei; @Grice:2009eea; @Findley:2013jf; @Costello:2009im, @Consortium:2012iz], but also support communities with distinct communication structures and propensities toward community instability. By understanding the healthy state of network structures across the human body, we will empower future studies to begin investigating how these community structures change in disease or otherwise altered states.

# Results

## Cohort Curation and Sample Processing
We studied the differences in microbiome phage-bacteria interaction networks across healthy human bodies by leveraging previously published sequence sets containing purified virome samples paired with bacterial metagenomes from whole metagenomic shotgun sequences. Our study contained three datasets, including a study of the impact of diet on the healthy human gut virome [@Minot:2011ez], the impact of anatomical location on the healthy human skin virome [@Hannigan:2015fz], and the viromes of monozygotic twins and their mothers [@Reyes:2010cwa; @Turnbaugh:2009ei]. The viromes associated with these datasets were subjected to virus-like particle (VLP) purification to eliminate other organismal DNA including bacteria, fungi, and humans, and allowed us to assess the actively replicating virome.

The publications we utilized were conducted over the span of five years using different methods and technologies, and therefore yielded different sequence abundances **(Supplemental Figure \ref{SequenceStats} A-B)**. The bacterial and viral sequences from those publications were quality filtered and assembled into contigs. Because contig assembly most commonly returns genome fragments, we clustered related bacteria and phage contigs by k-mer frequency and co-abundance using the CONCOCT algorithm. These clusters represent operationally defined units of related bacteria and phage genomes that we defined as operational genomic units (OGUs). These OGUs are conceptually similar to the operational taxonomic unit (OTU) and operational protein family (OPF) definitions used for grouping highly similar 16S rRNA gene and open reading frame sequences, respectively [@Schloss:2008ht]. The resulting contigs and OGUs demonstrated high sequence coverage and length **(Supplemental Figure \ref{ContigStats} - \ref{ClusterStats})**.

## Establishing a Model of Phage-Bacteria Interactions
We predicted which phage OGUs infected which bacteria using a random forest model trained on experimentally validated infectious relationships from six previous publications [@Jensen:1998vh;@Malki:2015tm;@Schwarzer:2012ez;@Kim:2012dh;@Matsuzaki:1992gw;@Edwards:2015iz]. This training set contained diverse bacteria and phages, with both broad and specific infectious ranges **(Supplemental Figure \ref{ValidationOverview} A - B)**. Phages with linear and circular genomes, as well as ssDNA and dsDNA genomes, were included in the analysis. Because this was a DNA sequencing study, RNA phages were not considered in the analysis **(Supplemental Figure \ref{ValidationOverview} C-D)**. This training set included both positive relationships (a phage infects a bacterium) and negative relationships (a phage does not infect a bacterium). This built on previous work, which focused only on positive relationships, by allowing us to validate the false positive *and* false negative rates associated with our candidate models [@Edwards:2015iz].

Four phage and bacterial genomic markers were used to predict infectious relationships between bacteria and phages: 1) genome nucleotide similarities, 2) gene amino acid sequence similarities, 3) CRISPR targeting of phages by bacterial CRISPR spacer sequences, and 4) similarity of protein families known to be associated with known protein-protein interactions [@Orchard:2014hq]. The resulting random forest model performed well, with an AUC of `r signif(genmodelper$ROC, digits = sigfig)`, a sensitivity of `r signif(genmodelper$Sens, digits = sigfig)`, and a specificity of `r signif(genmodelper$Spec, digits = sigfig)` **(Figure \ref{RocCurve} A)**. The most important predictor in the model was nucleotide similarity between genes, followed by nucleotide similarity of whole genomes **(Figure \ref{RocCurve} B)**. Protein family interactions were moderately important to the model, while CRISPRs were minimally important, likely because they were redundant with the blast information. Approximately one third of the relationships yielded no score and were therefore unknown, unable to be assigned an interaction **(Figure \ref{RocCurve} C)**.

We used our random forest model to classify the relationships between bacteria and phages in the experimental datasets. The relationships within the three studies were used to construct one master network, containing the three study sub-networks, which themselves each contained sub-networks for each sample **(Figure \ref{RocCurve} D)**. Metadata including study, sample ID, disease, and abundance within the community (based on sequence count) were also stored in the multi-study master network to allow for effective parsing and downstream analysis **(Supplemental Figure \ref{NetworkDiagram})**. The resulting master network was highly connected and contained `r prettyNum(genfigurestats[c(genfigurestats$class %in% "Total"),"Edges"], big.mark = ",")` infectious relationships among `r prettyNum(genfigurestats[c(genfigurestats$class %in% "Total"),"Vertices"], big.mark = ",")` nodes, `r prettyNum(nestats[c(nestats$cats %in% "PhageNodes"),"values"], big.mark = ",")` of which represented phages and `r prettyNum(nestats[c(nestats$cats %in% "BacteriaNodes"),"values"], big.mark = ",")` that represented bacteria. Although the network was highly connected, not all relationships were present in all samples. Furthermore, relationships were weighted by relative abundance of their associated bacteria and phages, meaning lowly abundant relationships could be present but insignificant compared to those that were highly abundant. Like the master network, the skin network exhibited a diameter of `r prettyNum(genfigurestats[c(genfigurestats$class %in% "SkinStudy"),"Diameter"], big.mark = ",")` (measure of graph size; the greatest number of traversed vertices required between two vertices) and included almost all of the master network nodes and edges **(Figure \ref{RocCurve} E - F)**. The gut diet and twin sample sets each contained fewer than 150 vertices, fewer than 20,000 relationships, and diameters of `r prettyNum(genfigurestats[c(genfigurestats$class %in% "DietStudy"),"Diameter"], big.mark = ",")`, suggesting more sparsely related phages and bacteria **(Figure \ref{RocCurve} E - F)**.

## Individuality of Microbial Networks
Skin and gut community membership and diversity are highly personal, with people remaining more similar to themselves over time compared to other people [@Grice:2009ee; @Hannigan:2015fz; @Minot:2013ih]. We determined whether this personal conservation extended to network structure using our microbiome network model. We calculated the degree of dissimilarity between each subject's network, based on phage and bacteria abundance, as well as centrality. Centrality was evaluated by first calculating the weighted eigenvector centrality of all bacteria and phages within each sample graph. Conceptually, this metric defines central phages as those that are highly abundant and infect many bacteria which themselves are abundant and infected by many other phages. Bacterial centrality was defined using the same metric. We then calculated the similarity of community networks using the weighted eigenvector centrality of all nodes between all samples. Biologically, samples with similar network structures are interpreted as having similar capacities to influence other microbes within the community, transmit genetic material (e.g. horizontal gene transfer), and maintain stability.

We used this network dissimilarity metric to test whether microbiome network structures were more similar within people over time compared to other people. We found that gut microbiome network structure clustered by person (ANOSIM p-value = `r signif(interstats[c(interstats$site %in% "DietAnosim"), "prob"], digits = sigfig)`, R = `r signif(interstats[c(interstats$site %in% "AS"), "prob"], digits = sigfig)`, **Figure \ref{intradiv} A**). We found evidence that network dissimilarity within people over the 8-10 day sampling period was less than the average dissimilarity between that person and others, although this was not statistically significant (p-value = `r signif(interstats[c(interstats$site %in% "Diet"), "prob"], digits = sigfig)`, **Figure \ref{intradiv} B**). The lack of statistical confidence was likely due to the small sample size of this dataset. Although there was evidence for gut network conservation among individuals, we found no evidence for conservation of gut network structures within families. The gut network structures were not more similar between twins and their mothers (intrafamily) compared to other families of twins and mothers (inter-family) (p-value = `r signif(interstats[c(interstats$site %in% "Skin"), "prob"], digits = sigfig)`, **Figure \ref{intradiv} C**).

Skin microbiome network structure was strongly conserved within individuals, and more adequately confirmed by our larger skin dataset (p-value = `r signif(interstats[c(interstats$site %in% "Twins"), "prob"], digits = sigfig)`, **Figure \ref{intradiv} D**). This distribution was similar when separated by anatomical sites, with most sites also being significantly more conserved within individuals **(Supplemental Figure \ref{allskin})**.

## Role of Diet & Obesity in Gut Microbiome Conectivity
Diet is a major environmental factor that influences resource availability and gut microbiome composition and diversity, including bacteria and phages [@Minot:2011ez; @Turnbaugh:2009hf; @David:2014cl]. Previous work in isolated culture-based systems has suggested that changes in nutrient availability are associated with altered phage - bacteria network structures, although this has yet to be tested in humans [@Poisot:2011jc]. We hypothesized that a change in diet would be associated with a change in network structure.

We evaluated the diet-associated differences in gut network structure by quantifying how central each sample's network was on average across all microbes. We accomplished this by utilizing two common centrality metrics: degree centrality and closeness centrality. **Degree centrality**, the simplest centrality metric, was defined as the number of connections each phage made to bacteria, or each bacterium made to phages. Because this metric alone offers only minimal insight, we supplemented it with measurements of closeness centrality. **Closeness centrality** is a metric of how close each phage or bacterium is to all of the other phages and bacteria in the network. A higher closeness centrality suggests that genetic information or the effects of altered abundance would be more impactful to all other microbes in the system. A networks with higher closeness centrality also indicates an overall greater degree of connections, which suggests a greater resilience to instability. Because these values are assigned to each phage and bacterium within each network, we calculated the average connectedness and corrected for the maximum potential degree of connectedness to obtain a single value for the connectedness of each graph.

We found that gut microbiome network structures associated with high fat diets were less connected than those of low fat diets **(Figure \ref{dietnetworks} A-B)**. Tests for statistical differences were not performed due to the small sample size. High fat diets exhibited less degree centrality **(Figure \ref{dietnetworks} A)**. This suggests that bacteria were targeted by less phages and that phage tropism was more specific. High fat diets also exhibited decreased closeness centrality **(Figure \ref{dietnetworks} B)**, indicating that microbes were more distant from other microbes in the community. This would make genetic transfer and altered abundance less capable of impacting other bacteria and phages within the network.

In addition to diet, there was also an association between obesity and network structure **(Figure \ref{dietnetworks} C-D)**. The obesity-associated network demonstrated a higher degree centrality **(Figure \ref{dietnetworks} C)** but a less closeness centrality, compared to the healthy controls **(Figure \ref{dietnetworks} D)**. This suggested that the obesity network was overall less connected, having microbes further from all other microbes within the community.

## Variation of Network Structure Across the Human Skin Landscape
Extensive work has illustrated differences in healthy human skin microbiome between anatomical sites, including bacteria, viruses, and fungi [@Grice:2009ee; @Findley:2013jf; @Hannigan:2015fz]. These communities vary by degree of skin moisture, oil, and environmental exposure. We hypothesized that like microbial composition and diversity, microbial network structure differed between anatomical sites. We tested this hypothesis by evaluating the changes in network structure between anatomical sites within our skin dataset.

We quantified the average centrality of each sample again using the weighted eigenvector centrality metric. We found that intermittently moist skin sites (dynamic sites that fluctuate between being moist and dry) were significantly less connected than the more stable moist and sebaceous environments (p-value < 0.001, **Figure \ref{skinnetwork} A)**. We also found that skin sites that were protected from the environment (occluded) were much more highly connected than those that were constantly exposed to the environment or only intermittently occluded (p-value < 0.001, **Figure \ref{skinnetwork} B)**.

We supplemented this analysis by comparing the network signatures using the centrality dissimilarity approach described above. The dissimilarity between samples was a function of shared relationships, degree of centrality, and bacteria/phage abundance. When using this supplementary approach, we found that network structures significantly clustered by moisture, sebaceous, and intermittently moist status **(Figure \ref{skinnetwork} C,E)**. We also found that occluded sites were significantly different from exposed and intermittently occluded sites, but there was no difference between exposed and intermittently occluded sites **(Figure \ref{skinnetwork} D,F)**.

# Discussion
We developed and implemented a network-based analytical approach that allowed us to begin evaluating the basic properties of the human microbiome through bacteria and phage *relationships*, instead of membership or diversity. The goal of this study was to provide an initial understanding of how phage-bacteria networks differ throughout the human body, so as to provide a baseline for future studies of how microbiome networks differ in disease states. This goal was similar to the initial studies of bacteria and viral diversity across the human body, as well as other environments [@Grice:2009eea; @Findley:2013jf; @Hannigan:2015fz; @Costello:2009im, @Consortium:2012iz; @Schloss:2005hz; @Minot:2011ez]. The use of network theory offered extensive analytical opportunities which could be applied to understand complex ecological communities. We focused on utilizing metrics of connectivity to understand the extent to which communities of bacteria and phages interact (e.g. horizontal gene transfer, modulated bacterial gene expression, alterations in abundance). By accomplishing this goal, we aimed to provide an initial understanding of human microbiome networks that will power fututure studies to use similar approaches in other setting such as disease states.

We reported that, just as gut microbiome and virome composition and diversity are conserved in individuals [@Hannigan:2015fz; @Grice:2009eea; @Findley:2013jf; @Minot:2013ih], gut and skin microbiome network structures were also conserved within individuals over time. Gut network structure was not conserved among family members. These findings suggest that microbiome properties such as stability and the potential for horizontal gene transfer are personal, and potentially impacted by personal factors ranging from the body's immune system to environemntal conditions such as climate and diet. The ability of environmental conditions to shape gut and skin microbiome network structure was supported by our finding that diet and skin location were assocaited with altered network structures.

We found evidence that diet was sufficient to alter gut microbiome network connectivity. Although our sample size was small, we found that high fat diets appeared to be less connected than low fat diets. This suggested that high fat diets may lead to less stable communities with a decreased ability to influence each other. We supported this finding with the observation that obesity is associated with decreased network connectivity. Further work will be required to characterize these relationships with a larger cohort, but our results allow us to speculate that the food we eat may not only impact what microbes colonize our guts, but may also impact interactions.

We also found evidence that skin microbiome network structure varied by skin environments. We observed that network structure differed between environmentally exposed and occluded skin sites. The sites under greater environmental fluctuation and exposure, the exposed and intermittently exposed sites, were less connected and therefore were expected to have a higher propensity for instability. Likewise, intermittently moist sites demonstrated less connectedness than the more stable moist and sebaceous sites. Together this suggested that body sites under greater degrees of fluctuation harbored less connected, potentially less stable microbiomes.

This work represents an initial step toward understanding the microbiome through its relationships. While these findings are informative, there are certainly caveats that should be noted. *First*, while our infection classification model is advantageous over existing models, we recognize that, like most classification models, there remains opportunity for improvement. For example, such a model is only as good as its training set, and future large-scale endeavors into infectious relationships (and the associated genomes) will provide more robust training and higher model accuracy. Just as we have improved on previous modeling efforts, we expect that new and creative scoring metrics will likely be integrated into this model to further improve model performance. *Second*, while informative, this work was done retrospectively and relied on published research from as many as seven years ago. These archived datasets were limited by the technology and costs of the time, meaning the datasets are poorly powered for the statistical analysis we strive for today. While we were able to present initial observations, follow-up studies will be required to validate our observations. Despite their limitations, these results will be important for informing design and interpretation of future studies of the human microbiome.

By showing that microbiome relationship structure, and therefore the community potential for dissemination of information and influence, as well as the potential for maintaining stability, differs significantly between body sites and environmental conditions we demonstrate the diversity of relationships across the human body. We found that other environmental factors, such as diet, also influence relationship structures. This information builds on previous work by suggesting that bacteria and phages not only preferentially colonize different body sites, but also interact differently, allowing for different communication structures and capacities for maintaining stability. This work will also provide a foundation for future studies to begin evaluating how microbiome network dynamics change in disease states, and how the information can be leveraged therapeutically.

# Materials & Methods

## Data Availability
All associated source code is available on GitHub at the following repository:

https://github.com/SchlossLab/Hannigan-2016-ConjunctisViribus.

## Data Acquisition & Quality Control
Raw sequencing data and associated metadata was acquired from the NCBI sequence read archive (SRA). Supplementary metadata was acquired from the same SRA repositories and their associated manuscripts. The gut virome diet study (SRA: `SRP002424`), twin virome studies (SRA: `SRP002523`; `SRP000319`), and skin virome study (SRA: `SRP049645`) were downloaded as `.sra` files. Sequencing files were converted to `fastq` format using the `fastq-dump` tool of the NCBI SRA Toolkit (v2.2.0). Sequences were quality trimmed using the Fastx toolkit (v0.0.14) to exclude bases with quality scores below 33 and shorter than 75 bp [@FASTXToolkit:wr]. Paired end reads were filtered to exclude and sequences missing their corresponding pair using the `get_trimmed_pairs.py` available in the source code.

## Contig Assembly
Contigs were assembled using the Megahit assembly program (v1.0.6) [@Li:2016kd]. A minimum contig length of 1 kb was used. Iterative k-mer stepping began at a minimum length of 21 and progressed by 20 until 101. All other default parameters were used.

## Contig Abundance Calculations
Contigs were concatenated into two master files prior to alignment, one for bacterial contigs and one for phage contigs. Sample sequences were aligned to phage or bacteria contigs using the Bowtie2 global aligner (v2.2.1) [@Langmead:2012jh]. We defined a mismatch threshold of 1 bp and seed length of 25 bp. Sequence abundance was calculated from the Bowtie2 output using the `calculate_abundance_from_sam.pl` script available in the source code.

## Operational Genomic Unit Binning
Contigs often represent large fragments of genomes. In order to reduce redundancy, and the resulting artificially inflated genomic richness within our dataset, it was important to bin contigs into operational units based on their similarity. This approach is conceptually similar to the clustering of related 16S rRNA sequences into operational taxonomic units (OTUs), although here we are clustering contigs into operational genomic units (OGUs) [@Schloss:2005hz].

We clustered contigs using the CONCOCT algorithm (v0.4.0) [@Alneberg:2014fc]. Because of our large dataset and limits in computational efficiency, we randomly subsampled the dataset to include 25% of all samples, and used these to inform contig abundance within the CONCOCT algorithm. CONCOCT was used with a maximum of 500 clusters, a k-mer length of four, a length threshold of 1 kb, 25 iterations, and exclusion of the total coverage variable.

OGU abundance (`Ao`) was obtained as the sum of the abundance of each contig (`Aj`) associated with that OGU. The abundance values were length corrected such that:

$$ { A }_{ O }=\frac { { 10 }^{ 7 }\sum _{ j=1 }^{ k }{ { A }_{ j } }  }{ \sum _{ j=1 }^{ k }{ { L }_{ j } }  } $$

Where `L` is the length of each contig `j` within the OGU.

## Open Reading Frame Prediction
Open reading frames (ORFs) were identified using the Prodigal program (V2.6.2) with the meta mode parameter and default settings [@Hyatt:2012cy].

## Classification Model Creation and Validation
The classification model for predicting interactions was built using experimentally validated bacteria-phage infections or validated lack of infections from six studies [@Jensen:1998vh;@Malki:2015tm;@Schwarzer:2012ez;@Kim:2012dh;@Matsuzaki:1992gw;@Edwards:2015iz]. Associated reference genomes were downloaded from the European Bioinformatics Institute (see details in source code). The model was created based on the four metrics listed below.

The four scores were used as parameters in a random forest model to classify bacteria and bacteriophage pairs as either having infectious interactions or not. The classification model was built using the Caret R package (v6.0.73) [@caretClassificatio:_U2Lit_1]. The model was trained using five-fold cross validation with ten repeats. Pairs without scores were classified as not interacting. The model was optimized using the ROC value. The resulting model performance was plotted using the plotROC R package.

##### Identify Bacterial CRISPRs Targeting Phages
CRISPRs were identified from bacterial genomes using the PilerCR program (v1.06) [@Edgar:2007bh]. Resulting spacer sequences were filtered to exclude spacers shorter than 20 bp and longer than 65 bp. Spacer sequences were aligned to the phage genomes using the nucleotide blast algorithm with default parameters (v2.4.0) [@Camacho:2009fc]. The mean percent identity for each matching pair was recorded for use in our classification model.

##### Detect Matching Prophages within Bacterial Genomes
Temperate bacteriophages infect and integrate into their bacterial host's genome. We detected integrated phage elements within bacterial genomes by aligning phage genomes to bacterial genomes using the nucleotide blast algorithm and a minimum e-value of 1e-10. The resulting bitscore of each alignment was recorded for use in our classification model.

##### Identify Shared Genes Between Bacteria and Phages
Phages may share genes with their bacterial hosts, providing us with evidence of phage-host infectious pairs. We identified shared genes between bacterial and phage genomes by assessing amino acid similarity between the genes using the Diamond protein alignment algorithm (v0.7.11.60) [@Buchfink:2015ki]. The mean alignment bitscores for each genome pair was recorded for use in our classification model.

##### Protein - Protein Interactions
The final method we used for predicting infectious interactions between bacteria and phages was by detecting pairs of genes whose proteins are known to interact. We assigned bacterial and phage genes to protein families by aligning them to the Pfam database using the Diamond protein alignment algorithm. We then identified which pairs of proteins were predicted to interact using the Pfam interaction information within the Intact database [@Orchard:2014hq]. The mean bitscores of the matches between each pair were recorded for use in our classification model.

## Virome Network Construction
The bacteria and phage operational genomic units (OGUs) were scored using the same approach as outlined above. The infectious pairings between bacteria and phage OGUs were classified using the random forest model described above. The predicted infectious pairings and all associated metadata was saved as a graph database using Neo4j graph database software (v2.3.1) [@Neoj:Fuwr6PBN]. This network was used for downstream community analysis.

## Centrality Analysis
Degree and closeness centrality were calculated using the associated functions within the igraph R package (v1.0.1) [@Theigraphsoftware:vh]. Briefly, the **closeness centrality** of Vi was calculated taking the inverse of the average length of the shortest path (`d`) between nodes Vi and the `k` other nodes Vj, such that closeness centrality of node Vi is:

$$ { C }_{ C }\left( { V }_{ i } \right) ={ \left( \sum _{ j=1 }^{ k }{ d\left( { V }_{ i },{ V }_{ j } \right)  }  \right)  }^{ -1 } $$

The distance between nodes (`d`) was calculated as the shortest number of edges required to be traversed to move from one node to another. 

The simple metric of **degree centrality** of node Vi was defined as the sum of the number of k edges Ei associated with that node, such that:

$$ { C }_{ D }\left( { V }_{ i } \right) =\sum _{ i=1 }^{ k }{ { E }_{ i } } $$

The Eigenvector centrality was calculated using the values of the first eigen vector of the associated adjacency matrix that are associated with each vertex `Vi`. Conceptually, this function results in a centrality value that reflects the connections of the vertex, as well as the centrality of the connected vertices.

The **centralization** metric was used to assess the average centrality of each sample graph `G`. Centralization was calculated by taking the sum of each vertex `Vi` centrality from the graph maximum centrality `Cw`, such that:

$$ C\left( G \right) =\frac { \sum _{ i=1 }^{ k }{ \max _{ w }{ c\left( w \right)  } -c\left( { v }_{ i } \right)  }  }{ { T } }  $$

The values were corrected for uneven graph sizes by dividing the centralization score by the maximum theoretical centralization (`T`) for a graph with the same number of vertices.

## Network Relationship Dissimilarity
We assessed similarity between graphs by evaluating the shared centrality of their vertices, as has been done previously. More specifically, we calculated the dissimilarity between graphs `Gi` and `Gj` using the Bray-Curtis dissimilarity metric and Eigenvector centrality values such that:

$$ B\left( { G }_{ i },{ G }_{ j } \right) =1-\frac { 2{ C }_{ ij } }{ { C }_{ i }+{ C }_{ j } } $$

Where `Cij` is the sum of the lesser centrality values for those vertices shared between graphs, and `Ci` and `Cj` are are the total number of vertices found in each graph. This allows us calculate the dissimilarity between graphs based on the shared centrality values between the two graphs.

# Acknowledgments
We thank the members of the Schloss lab for their underlying contributions. GDH is supported in part by the University of Michigan Molecular Mechanisms of Microbial Pathogenesis Fellowship.

# Disclosure Declaration
The authors report no conflicts of interest.

\newpage

# Figures

![**Summary of Multi-Study Network Model.** *(A) ROC curve resulting from ten iterations used to create the phage - bacteria infection prediction model. (B) Importance scores associated with the metrics used in the random forest model to predict relationships between bacteria and phages. The importance score is defined as the mean decrease in accuracy of the model when a feature (e.g. Pfam) is excluded. (C) Proportions of samples included (gray) and excluded (red) in the model. Samples were excluded from the model because they did not yield any scores. Those interactions without scores were defined as not having interactions. (D) Bipartite visualization of the resulting phage-bacteria network. This network includes information from all three published studies. (E) Network diameter (measure of graph size; the greatest number of traversed vertices required between two vertices), (F) number of vertices, and (G) number of edges (relationships) for the total network (yellow) and the individual study sub-networks (diet study = red, skin study = green, twin study = orange).* \label{RocCurve}](../figures/rocCurves.pdf){ width=90% }

![**Intrapersonal vs Interpersonal Network Dissimilarity Across Different Human Systems.** *(A) NMDS ordination illustrating network dissimilarity between subjects over time. Each sample is colored by subject, with each sample pair collected 8-10 days apart. Dissimilarity was calculated using the Bray-Curtis metric based on abundance weighted Eigenvector centrality signatures, with a greater distance representing greater dissimilarity in bacteria and phage centrality and abundance. (B) Quantification of gut network dissimilarity within the same subject over time (intrapersonal) and the mean dissimilarity between the subject of interest and all other subjects (interpersonal). The p-value is also provided. (C) Quantification of gut network dissimilarity within subjects from the same family (intrafamily) and the mean dissimilarity between subjects within a family and those of other families (interfamily). The p-value is also provided. (D) Quantification of skin network dissimilarity within the same subject and anatomical location over time (intrapersonal) and the mean dissimilarity between the subject of interest and all other subjects at the same time and the same anatomical location (interpersonal). The p-value is also provided. *\label{intradiv}](../figures/intrapersonal_diversity.pdf){ width=90% }

![**Impact of Diet and Obesity on Gut Network Structure.** *(A) Quantification of average degree centrality (number of edges per node) and (B) closeness centrality (average distance from each node to every other node) of gut microbiome networks of subjects limited to exclusively high fat or low fat diets. Lines represent the mean degree of centrality for each diet. (C) Quantification of average degree centrality and (D) closeness centrality between obese and healthy adult women.*\label{dietnetworks}](../figures/dietnetworks.pdf){ width=90% }

![**Impact of Skin Micro-Environment on Microbiome Network Structure.**  *(A) Notched box-plot depicting differences in average Eigenvector centrality between moist, intermittently moist, and sebaceous skin sites and (B) occluded, intermittently occluded, and exposed sites. Notched box-plots were created using ggplot2 and show the median (center line), the inter-quartile range (IQR; upper and lower boxes), the highest and lowest value within 1.5 * IQR (whiskers), outliers (dots), and the notch which provides an approximate 95% confidence interval as defined by 1.58 * IQR / sqrt(n). (C) NMDS ordination depicting the differences in skin microbiome network structure between skin moisture levels and (D) occlusion. Samples are colored by their environment and their dissimilarity to other samples was calculated as described in figure \ref{intradiv}. (E) The statistical differences of networks between moisture and (F) occlusion status were quantified with an anova and post hoc Tukey test. Cluster centroids are represented by dots and the extended lines represent the associated 95% confidence intervals. Significant comparisons (p-value < 0.05) are colored in red, and non-significant comparisons are gray.*\label{skinnetwork}](../figures/skinplotresults.pdf){ width=75% }

\newpage

# Supplemental Figures

\beginsupplement

![**Sequencing Depth Summary.** *Number of sequences that aligned to (A) Phage and (B) Bacteria operational genomic units. Sequencing count aligned to OGUs was length corrected.*\label{SequenceStats}](../figures/SequenceAbund.pdf){ width=80% }

\newpage

![**Contig Summary Statistics.** *Scatter plot heat map with each hexagon representing an abundance of contigs. Contigs are organized by length on the x-axis and the number of aligned sequences on the y-axis.*\label{ContigStats}](../figures/ContigStats.pdf){ width=50% }

\newpage

![**Operational Genomic Unit Summary Statistics.** *Scatter plot with operational genomic unit clusters organized by average contig length within the cluster on the x-axis and the number of contigs in the cluster on the y-axis. Operational genomic units of (A) bacteriophages and (B) bacteria are shown.*\label{ClusterStats}](../figures/ClusterStats.pdf){ width=80% }

\newpage

![**Summary information of validation dataset used in the interaction predictive model.** *A) Categorical heat-map highlighting the experimentally validated positive and negative interactions. Only bacteria species are shown, which represent multiple reference strains. Phages are labeled on the x-axis and bacteria are labeled on the y-axis. B) World map illustrating the sampling locations used in the study (red dots). C) Quantification of bacterial host strains known to exist for each phage. D) Genome strandedness and E) linearity of the phage reference genomes used for the dataset.*\label{ValidationOverview}](../figures/BenchmarkDataset.pdf){ width=100% }

\newpage

![**Structure of the interactive network.** *Metadata relationships to samples (Phage Sample ID and Bacteria Sample ID) included the associated time point, the study, the subject the sample was taken from, and the associated disease. Infectious interactions were recorded between phage and bacteria operational genomic units (OGUs). Sequence count abundance for each OGU within each sample was also recorded.*\label{NetworkDiagram}](../figures/graphdatabasediagram.pdf){ width=50% }

\newpage

![**Intrapersonal vs Interpersonal Dissimilarity of the Skin.** *Quantification of skin network dissimilarity within the same subject and anatomical location over time (intrapersonal) and the mean dissimilarity between the subject of interest and all other subjects at the same time and the same anatomical location (interpersonal), separated by each anatomical site (forehead [Fh], palm [Pa], toe web [Tw], umbilicus [Um], antecubital fossa [Ac], axilla [Ax], and retroauricular crease [Ra]). Below is the probability distribution of the changes between intrapersonal and interpersonal diversity, representing the probability that intrapersonal dissimilarity will be lower than interpersonal dissimilarity (intrapersonal change less than zero). The probability that the slope will be less than zero (integral from negative infinity to zero) is provided in the top left corner.*\label{allskin}](../figures/intraallskin.pdf){ width=75% }

\newpage

# References
